Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +203,209 @@ protected override bool IsValidController(IMixedRealityController controller)
+                case SolverSafeZone.BelowWrist:
+                    return jointedHand.TryGetJoint(TrackedHandJoint.Wrist, out referenceJointPose1) ? referenceJointPose1.Position : Vector3.zero;
+
+                case SolverSafeZone.RadialSide:
+                    referenceJoint1 = TrackedHandJoint.IndexKnuckle;
+                    referenceJoint2 = TrackedHandJoint.ThumbProximalJoint;
+                    break;
+                case SolverSafeZone.UlnarSide:
+                default:
+                    referenceJoint1 = TrackedHandJoint.IndexKnuckle;
+                    referenceJoint2 = TrackedHandJoint.ThumbProximalJoint;
+                    break;
+            }
+
+
+            if (!jointedHand.TryGetJoint(referenceJoint1, out referenceJointPose1) ||
+                !jointedHand.TryGetJoint(referenceJoint2, out referenceJointPose2))
+            {
+                return Vector3.zero;
+            }
+
+
+            return Vector3.Lerp(referenceJointPose1.Position, referenceJointPose2.Position, .5f);
+        }
+
+        /// <summary>
+        /// Coroutine function that's invoked when the attached object becomes world-locked. It uses the 
+        /// logical checks invoked during IsValidController to determine whether the menu should reattach
+        /// to the hand or not.
+        /// </summary>
+        /// <returns></returns>
+        private IEnumerator WorldLockedReattachCheck()
+        {
+            while (!SolverHandler.UpdateSolvers && useGazeActivation)
+            {
+                MixedRealityPose palmPose;
+                var jointedHand = GetController(SolverHandler.CurrentTrackedHandedness) as IMixedRealityHand;
+                if (jointedHand != null)
+                {
+                    if (jointedHand.TryGetJoint(TrackedHandJoint.Palm, out palmPose))
+                    {
+                        float palmCameraAngle = Vector3.Angle(palmPose.Up, CameraCache.Main.transform.forward);
+                        if (IsPalmMeetingThresholdRequirements(jointedHand, palmPose, palmCameraAngle) &&
+                            IsUserGazeMeetingThresholdRequirements(jointedHand))
+                        {
+                            gazeActivationAlreadyTriggered = false;
+                            SolverHandler.UpdateSolvers = true;
+                        }
+                    }
+                }
+
+                yield return null;
+            }
+        }
+
+
+        #region MonoBehaviour Implementation
+
+        /// <summary>
+        /// When enabled, ensure that there are no outlying status changes that would prevent HandConstraintPalmUp from 
+        /// properly working (like gazeActivationAlreadyTriggered being set to true previously, or SolverHandlers.UpdateSolvers being false)
+        /// </summary>
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            gazeActivationAlreadyTriggered = false;
+            SolverHandler.UpdateSolvers = true;
```

Comment:
@KenJakubzak I would actually expect that if I called `SolverHandler.UpdateSolvers = false` before disabling a component, then on enable, that component would not all of a sudden call `SolverHandler.UpdateSolvers = true` for me. I think this was to solve a specific case in the Hand Menu examples scene where switching menus could cause menus to be stuck in space. Perhaps this should instead be implemented in a script scoped just to the example scene?

Reply:
I can do this.

Reply:
Actually; was thinking about this some more. This will happen whenever the object is disabled while running the coroutine. When reenabled, it simply won't reconnect due to it turning off the solver when deattaching from the hand (and the activation of the UI wouldn't necessarily lead to it having the solver update being turned back on; leading to the UI then being functionally broken again).

I think our sample scene exposes this, but it isn't specific to just this scene as developers could hit this with their own app.

We could use another script to do this management as you; but at the same time, that means we have a component that could still end up in this state. 

Would a potential option be that we add this as an option the developer can select (through a publicly-facing bool)?

Reply:
Actually; thinking about it some more. Fair point. We can hook onto the events that the buttons trigger and set these values when there's a menu switch in the scene. Will implement it now.